app/javascript/controllers/refine/modal-controller.js

```diff
@@ -0,0 +1,41 @@
+import { Controller } from "@hotwired/stimulus"
+import { useClickOutside } from 'stimulus-use'
+
+// simple controller to hide/show the filter modal
+export default class extends Controller {
+  static targets = ["frame"]
+
+  static values = {
+    src: String,
+    isOpen: {type: Boolean, default: false}
+  }
+
+  connect() {
+    console.log("Connecting modal controller")
+    useClickOutside(this)
+  }
+
+  disconnect() {
+  }
+
+  open(event) {
+    console.log("Opening modal!!!", this.srcValue)
```

Comment:
Intentionally leaving this for now. Will clean it up before this feature is fully released